src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1464,14 +1467,22 @@ HasUnresolvedExternParamsWalker(Node *expression, ParamListInfo boundParams)
 		if (boundParams && paramId > 0 && paramId <= boundParams->numParams)
 		{
 			ParamExternData *externParam = &boundParams->params[paramId - 1];
+			Oid paramType = externParam->ptype;
```

Comment:
@mtuncer @jasonmp85 

This line causes an invalid memory read error on Valgrind because when `paramFetch ` is supplied, length of the `params []` could be zero, as documented in the Postgres code.

So, we shouldn't be accessing `externParam ` at this point.

It looks like a very similar line was already in Citus long before PG 11, see [commit](https://github.com/citusdata/citus/commit/6939cb8c561f896ece0a28b48cae7d8cb9f26ed3).  

Do you remember whether this is related with PG 11 or not? (e.g., was it always true that when `paramFetch` exists, the array size could be zero.)

Here is the valgrind error:
```
==31715== VALGRINDERROR-BEGIN
==31715== Invalid read of size 4
==31715==    at 0x6040559: HasUnresolvedExternParamsWalker (distributed_planner.c:1567)
==31715==    by 0x3E2A49: expression_tree_walker (nodeFuncs.c:2087)
==31715==    by 0x604050F: HasUnresolvedExternParamsWalker (distributed_planner.c:1601)
==31715==    by 0x3E2ACB: expression_tree_walker (nodeFuncs.c:2119)
==31715==    by 0x604050F: HasUnresolvedExternParamsWalker (distributed_planner.c:1601)
==31715==    by 0x3E2F4C: query_tree_walker (nodeFuncs.c:2263)
==31715==    by 0x6040599: HasUnresolvedExternParamsWalker (distributed_planner.c:1594)
==31715==    by 0x6041660: CreateDistributedPlannedStmt (distributed_planner.c:490)
==31715==    by 0x6041988: distributed_planner (distributed_planner.c:158)
==31715==    by 0x4404D3: planner (planner.c:260)
==31715==    by 0x50032E: pg_plan_query (postgres.c:832)
==31715==    by 0x50046D: pg_plan_queries (postgres.c:898)
```

Reply:
Ok, I found the Postgres commit that changed the behaviour: https://github.com/postgres/postgres/commit/6719b238e8f0ea83c39d2b1728c7670cdaa34e06

I opened a PR (#2457) to reflect the change and make sure that we don't have invalid memory access.